orthoinference/src/main/java/org/reactome/orthoinference/InferEvents.java

```diff
@@ -0,0 +1,307 @@
+package org.reactome.orthoinference;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.GKSchemaAttribute;
+import org.gk.schema.SchemaClass;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.reactome.release.common.database.InstanceEditUtils;
+
+/**
+ *
+ * @author jcook
+ * 
+ * The Java version of infer_events.pl -- The gist of this module is that it looks at all existing Human ReactionlikeEvent (RlE) instances (mostly Reactions and BlackBoxEvents) in the Test_Reactome database,
+ * and attempts to computationally infer them in each of Reactome's model organisms. Each RlE is broken down into its primary components (input, output, catalyst, and regulator), which are themselves broken
+ * into their PhysicalEntity subunits. The homology data used for the inference process currently comes from Ensembl Compara and is generated during the 'Orthopairs' step of the Reactome release process.
+ * After all inference attempts for each RlE has been completed in an organism, the pathways that contain the reactions are filled with these newly inferred ones. 
+ * 
+ *
+ */
+
+// To Reviewers: I recommend following the code sequentially, rather then looking at each class separately.
+public class InferEvents
+{
+	static MySQLAdaptor dbAdaptor = null;
+	private static GKInstance instanceEdit;
+	private static GKInstance speciesInst;
+	private static HashMap<GKInstance,GKInstance> manualEventToNonHumanSource = new HashMap<GKInstance,GKInstance>();
+	private static ArrayList<GKInstance> manualHumanEvents = new ArrayList<GKInstance>();
+
+	@SuppressWarnings("unchecked")
+	public static void eventInferrer(Properties props, String pathToConfig, String pathToSpeciesConfig) throws Exception
```

Comment:
pathToConfig doesn't look like it is used -- can it be removed?

Reply:
Yep, InferEvents used to be the Main class for this script and when I moved the 'Main' to its own class, I kept the arguments the same